wandb/apis/importers/internals/internal.py

```diff
@@ -0,0 +1,386 @@
+import json
+import logging
+import math
+import os
+import queue
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Dict, Iterable, Optional
+
+import numpy as np
+from google.protobuf.json_format import ParseDict
+from tenacity import retry, stop_after_attempt, wait_random_exponential
+
+from wandb import Artifact
+from wandb.proto import wandb_internal_pb2 as pb
+from wandb.proto import wandb_settings_pb2
+from wandb.proto import wandb_telemetry_pb2 as telem_pb
+from wandb.sdk.interface.interface import file_policy_to_enum
+from wandb.sdk.interface.interface_queue import InterfaceQueue
+from wandb.sdk.internal import context
+from wandb.sdk.internal.sender import SendManager
+from wandb.sdk.internal.settings_static import SettingsStatic
+from wandb.util import coalesce, recursive_cast_dictlike_to_dict
+
+from .protocols import ImporterRun
+
+ROOT_DIR = "./wandb-importer"
+
+# silences the internal messages; set to INFO or DEBUG to see them
+handlers = []
+if os.getenv("WANDB_IMPORTER_ENABLE_RICH_LOGGING"):
+    from rich.logging import RichHandler
+
+    handlers = [RichHandler(rich_tracebacks=True, tracebacks_show_locals=True)]
+    logging.basicConfig(level=logging.WARN, handlers=handlers)
+else:
+    logging.basicConfig(level=logging.WARN)
+
+logging.getLogger("urllib3").setLevel(logging.ERROR)
+logging.getLogger("urllib3.connectionpool").setLevel(logging.ERROR)
```

Comment:
The file/artifacts uploading internals import `urllib3` and it can get noisy with a lot of warnings that distract from actual warnings the importer is trying to warn you about